executing/executing.py

```diff
@@ -536,15 +588,19 @@ def compile_similar_to(source, matching_code):
 sentinel = 'io8urthglkjdghvljusketgIYRFYUVGHFRTBGVHKGF78678957647698'
 
 def is_rewritten_by_pytest(code):
+    # type: (types.CodeType) -> bool
     return any(
         bc.opname != "LOAD_CONST" and isinstance(bc.argval,str) and bc.argval.startswith("@py")
         for bc in get_instructions(code)
     )
 
 
 class SentinelNodeFinder(object):
+    result = None # type: EnhancedAST
+
     def __init__(self, frame, stmts, tree, lasti, source):
-        assert_(stmts)
+        # type: (types.FrameType, Union[List[EnhancedAST], Set[EnhancedAST]], ast.AST, int, object) -> None
```

Comment:
I don't think stmts should be a list sometimes.

`tree` should probably be a Module, then compile_similar_to doesn't need casts to call. Or maybe compile_similar_to should just accept AST.

Shouldn't `source` be `Source`?

Reply:
Agreed. IIRC stmts must have been a List sometimes at one point. I hadn't given a proper type to `source` because nothing in the function uses it, but I've now aligned the types here and for `PositionNodeFinder`